### PR TITLE
WSO-192 - CKEditor - updating changes in brick

### PR DESCRIPTION
**Pull request by SandyVallee**
added hook_js _alter to swap js file from ckeditor with a patched version.
Here is a link to the one line change. Lines 6 and 203 changed.
https://github.com/ucalgary/www.ucalgary/pull/816/files#diff-14f3bbfa71adfa8473fb057e317e40a1R203

### CODE REVIEW

File: sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
 /*
 Copyright (c) 2003-2011, CKSource - Frederico Knabben. All rights reserved.
 For licensing, see LICENSE.html or http://ckeditor.com/license
+Sandy Vallee - testing patch https://www.drupal.org/files/ckeditor-1990224-2_0.patch - July 6th 2017
```
Comment (taeram) [inline sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js]: Please remove unnecessary comment.

File: sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 */
 if (typeof window.CKEDITOR_BASEPATH === 'undefined') {
   window.CKEDITOR_BASEPATH = Drupal.settings.ckeditor.editor_path;
+  delete Drupal.ckeditorInstance;//added from patch https://www.drupal.org/files/ckeditor-1990224-2_0.patch
```
Comment (taeram) [inline sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js]: Please remove comment from this line, as git tracks why code is added, so there's no need to do it in the code as well.

File: sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js [hunk 3]
```diff
@@ -199,7 +201,8 @@ if (typeof window.CKEDITOR_BASEPATH === 'undefined') {
       delete Drupal.ckeditorInstance;
 
     $("#" + textarea_id).val(CKEDITOR.instances[textarea_id].getData());
-    CKEDITOR.instances[textarea_id].destroy(true);
+    //CKEDITOR.instances[textarea_id].destroy(true);//remove this line for the patch https://www.drupal.org/files/ckeditor-1990224-2_0.patch
```
Comment (taeram) [inline sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js]: Commented out code should never be left in a project, as it reduces readability of the code. Also, any code that is removed can always be retrieved by going through the commit history in git.

Please remove this line of commented out code.

File: sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js [hunk 4]
```diff
@@ -199,7 +201,8 @@ if (typeof window.CKEDITOR_BASEPATH === 'undefined') {
...
+    delete(CKEDITOR.instances[textarea_id]); //added from patch https://www.drupal.org/files/ckeditor-1990224-2_0.patch
```
Comment (taeram) [inline sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js]: As explained above, please remove comment from this line as well.

File: sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js (hunk 1 above)
Comment (SandyVallee) [inline sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js]: done

File: sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js (hunk 2 above)
Comment (SandyVallee) [inline sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js]: Done.  I just found that going through the other developers work with vague comments like "this module has been patched" was unhelpful when knowing what patchs had been applied in the past. Dan suggested that it was not a bad idea to include the link to the patch added at the top of the file comments.  Anyway, I will remove and just put the patch link in these git comments.

File: sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js (hunk 3 above)
Comment (SandyVallee) [inline sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js]: Ok done. Again this was not meant to be merged into main branch, it was just for ease of use when removing code from this file and converting in a hook file for a final fix.

File: sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js (hunk 4 above)
Comment (SandyVallee) [inline sites/all/modules/contrib/ckeditor/includes/ckeditor.utils.js]: done

**Comment by taeram** [comment]
Closing as ticket has been cancelled

**Pull request closed without merging**